components/scream/src/physics/shoc/atmosphere_macrophysics.hpp

```diff
@@ -328,22 +401,26 @@ class SHOCMacrophysics : public scream::AtmosphereProcess
     view_1d_const phis;
 
     // Assigning local variables
-    void set_variables(const int ncol_, const int nlev_,
+    void set_variables(const int ncol_, const int nlev_, const int num_qtracers_,
+                       const view_1d_int_const& convert_wet_dry_idx_d_,
                        const view_2d_const& rrho_,
                        const view_2d& qv_, const view_2d_const& qw_, const view_2d& qc_, const view_2d_const& qc_copy_,
-                       const view_2d& tke_, const view_2d_const& tke_copy_, const view_2d_const& qc2_,
+                       const view_2d& tke_, const view_2d_const& tke_copy_, const view_3d& qtracers_, const view_2d_const& qc2_,
                        const view_2d& cldfrac_liq_, const view_2d& inv_qc_relvar_,
                        const view_2d& T_mid_, const view_2d_const& dse_, const view_2d_const& z_mid_, const view_1d_const phis_)
     {
       ncol = ncol_;
       nlev = nlev_;
+      num_qtracers = num_qtracers_;
+      convert_wet_dry_idx_d = convert_wet_dry_idx_d_;
       rrho = rrho_;
       qv = qv_;
       qw = qw_;
       qc = qc_;
       qc_copy = qc_copy_;
       tke = tke_;
       tke_copy = tke_copy_;
+      qtracers = qtracers_;
```

Comment:
Not a perf/mem issue, but it is somewhat confusing to see qtracers as well as qv/qw/qc as inputs to the function. You can of course store a copy of qc/qv/qw, but you might just grab qv/qw/qc from qtracers. Of course, you would have to pass the indices of qv/qw/qc in order to properly subview qtracers.

Note: this is not a _big_ deal. It's just that when I see `f(x,y)` i always think that `x` and `y` are _independent_ variables. If `x` _has_ to be a subview of `y`, then it's probably more self-documenting to have `f(idx_x,y)`, and extract `x=y(:,idx_x,:)` inside the function.

Reply:
@bartgol , do you still have this comment?